packages/core/src/workflows/workflow.ts

```diff
@@ -1190,8 +1244,14 @@ export class Workflow<
     return this.#serializedStepSubscriberGraph;
   }
 
-  get steps() {
-    return this.#steps;
+  get steps(): Record<string, StepAction<string, any, any, any>> {
+    return Object.entries(this.#steps).reduce(
```

Comment:
can you explain what you're doing here? You're cloning the object?

Reply:
I changed the interface of the private `this.#steps` property, however there are places outside of core depending on the former structure through the getter. 

This change here is building out the expected structure.

You'll notice I'm picking the `step` member from the main step:

```ts
(acc, [key, step]) => {
  acc[key] = step.step;
  return acc;
},
```